etc/docs/Design.md

```diff
@@ -0,0 +1,100 @@
+# Things to cover
+
+## API
+
+### High level Library API
+
+The library API should provide the following methods to the clients:
+
+* Initialize an executor instance with the following settings
+  * Limits for resources
+    
+
+* Execute a command with the supplied arguments, that returns a pointer to a `Command` object that can be later used to manipulate the underlying process
+
+### GRPC API definition
+
+The GRPC API defines the following methods:
+
+* ExecuteCommand: Starts a command with limited resources in its own namespace with the supplied arguments
+* StopExecution : Stop a certain execution 
+* ListExecutions: List all the executions the connected client has rights to
+* GetExecution: Gets metadata about a certain execution defined by an execution ID
+* GetOutStream: Get a stream of lines of standard output (stdout & stderr) of a certain execution
+* GetOut: Get all lines of standard output (stdout & stderr) that has been written so far by the process
+
+To read about the exact details of the API you can look at [the proto file](api.proto)
+
+### Streaming data
+
+In order to be able to stream and also send the full output of the processes we need to store the output of this in memory for simplicty, in order to be able to keep memory from being overloaded a simple recylcing of these processes will be implemented. See [Tradeoffs](#tradeoffs) section for information about why this solution should not be used in production  
+
+## Security
+
+### Cipher suite
+
+In order to be secure the app will use the TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 cypher suite.
```

Comment:
What about TLS version?

Reply:
This cipher implies TLS 1.2, since it uses AES (not available before 1.2) and RSA (not available in 1.3). 

I used TLS 1.2 since, it's more widely used still, but after re-reading the specification it explicitly asks for the "strongest transport encryption" therefore I think I will select a TLS 1.3 compatible cipher suite